src/schema.jl

```diff
@@ -366,37 +368,67 @@ julia> 1 .- d.a .* d.b
  -2.9996186355944543
 ```
 """
-unprotect(::Protected{Ctx}) where Ctx = Ctx
+unprotect(::Protected{Ctx}) where {Ctx} = Ctx
 
 function apply_schema(t::FunctionTerm{typeof(unprotect)}, schema::Schema, Ctx::Protected)
     tt = only(t.args)
     apply_schema(tt, schema, unprotect(Ctx))
 end
 
-# un-protecting special syntax: &, +, and *
-# 
-# if these occur as children of a non-special call, they are protected by
-# default.  so if we encounter them during apply_schema and we're NOT
-# in a Protected context, we need to call the corresponding op on the
-# # arguments of the FunctionCall
-for op in (+, &, *)
-    @eval begin
-        apply_schema(t::FunctionTerm{typeof($op)}, sch::Schema, Mod::Type) =
+"""
+    StatsModels.@unprotect f
+
```

Comment:
`@unprotect` may not be the best name, as usually `@func x` is a convenience equivalent of `func(x)`, which isn't the case here. Maybe something like `@support_unprotect` or even `@special_term`?

Also:
```suggestion
    StatsModels.@unprotect f

Generate methods necessary for function `f` to support [`unprotect`](@ref).

```

Reply:
yeah that's a good point, I hadn't thought about how there was already a function called `unprotect`.  I think `@support_unprotect` is a good name.